### DISCUSSION TITLE
Base table (linkace.users) not found after attempted repair

### DISCUSSION THREAD

**Discussion by stephenmetzger** [Kovah/LinkAce#447] [Support]
I have LinkAce installed with docker on UnRAID. I tried to access my linkace instance today, and was routed to a screen to re-configure - followed by a warning that my database already contained data and that proceeding would delete. So, i grabbed a backup of my mariaDB database for linkace, and the appdata folder for linkace and proceeded with re-configuration. 

After reconfiguration, I replaced the mariadb database folder with my backed-up copy, and am now getting an 
`Error 500 - Server Error (An internal server error occured. If you are the administrator, consult the application logs for details.).`

Logs show: 
` Base table or view not found: 1932 Table 'linkace.users' doesn't exist in engine at /app/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:82)
[stacktrace]
`
My mariadb shows my database 'linkace' with the table 'users'. How can I recover from this?

**Comment by Kovah** [comment]
Did you do an update before accessing LinkAce? Would like to know if this happened because of version 1.10.
Besides, make sure that the database is restored correctly. It seems that authentication was successful but the database is just empty.

**Reply by stephenmetzger** [reply]
Yes, this occurred after an update. 

Dug a little deeper, and although tables still show in my mariadb database:
`MariaDB [linkace]> show tables;
+-----------------------+
| Tables_in_linkace     |
+-----------------------+
| failed_jobs           |
| jobs                  |
| link_lists            |
| link_tags             |
| links                 |
| lists                 |
| migrations            |
| model_has_permissions |
| model_has_roles       |
| notes                 |
| password_resets       |
| permissions           |
| revisions             |
| role_has_permissions  |
| roles                 |
| settings              |
| tags                  |
| users                 |
+-----------------------+
18 rows in set (0.001 sec)
`

Opening up the database in phpmyadmin shows (for all tables): ` 'linkace.<TABLE>' doesn't exist in engine`. 

To backup, I had just copied the linkace directory in my mariadb appdata folder, and then pasted it back in. Thinking that may not have been enough.... Any suggestions? I literally just transferred all my bookmarks from various machines/webbrowsers into LinkAce to consolidate, and now its looking like I've lost everything....

**Reply by Kovah** [reply]
Data should definitely not be deleted, that's why there is a warning that all data would be deleted in the setup. But I have no idea why the setup popped up after the update.

In general, never use the appdata folder of any database for backups. Seems there was something broken in that process. You can try to follow these steps to repair the database, but please only use sql dumps for backups in the future.
- https://stackoverflow.com/questions/25039927/database-corruption-with-mariadb-table-doesnt-exist-in-engine
- https://www.dba-ninja.com/2020/07/how-to-fix-table-doesnt-exist-in-engine-error-for-mariadb-error-1932.html

**Comment by stephenmetzger** [comment]
Got very lucky and was able to recover by restoring my entire appdata folder in UnRAID. In hindsight, I had also exported my bookmarks to HTML and CSV as I was migrating and could have painstakingly re-created by database... but glad I didn't have to. Lesson learned, looking into SQL dumps for my databases. I appreciate your responsiveness. Any other recommendations for backing up docker-installed LinkAce instances for easy recovery?

**Reply by Kovah** [reply]
Glad you were able to restore the database.
If you have any AWS S3 compatible storage at hand (Backblaze being a good provider, or Minio for self hosting), you can use the application backup feature: https://www.linkace.org/docs/v1/configuration/application-backups/

**Comment by stephenmetzger** [comment]
Reopening this because it happened again. I have not restored anything yet, but again, when trying to access LinkAce, i get directed to the setup screen, and after filling in my mariadb linkace info, I get the message 

```Caution! We found data in the database you specified! Please make sure that you have a backup of that database and confirm the deletion of all data.```

Why is this happening, and what is the best way to fix?

**Reply by Kovah** [reply]
I don't know why this happens again and again. I am running LinkAce on two servers installed with Docker for more than two years now, without any issues upgrading them. If tables randomly disappear, I guess there's something wrong with the database, because there is no login in LinkAce that could randomly delete them.

**Reply by stephenmetzger** [reply]
Odd, I haven't proceeded with setup, but exploring the tables in my linkace database, tables appear to be intact, its just seem that linkace isn't pickup it up on start up. 

Any logs or anything I could try to grab for you that could provide insight?

Last lines in the LinkAve logs is ```[2022-04-07 01:23:32] production.ERROR: SQLSTATE[42S02]: Base table or view not found: 1932 Table 'linkace.settings' doesn't exist in engine (SQL: select * from `settings` where `user_id` is null) ```

I see the table linkace,settings in my linkace db, with 35 records. All seems to be there.

**Reply by martadinata666** [reply]
I think this either your `.env` overwritten or forgot to `php artisan migrate`.
